src/backend/api/Fusion.Resources.Domain/Queries/GetPersonRelevantOrgUnits.cs

```diff
@@ -0,0 +1,383 @@
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Fusion.AspNetCore.OData;
+using Fusion.Services.LineOrg.ApiModels;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Fusion.Resources.Domain.Queries
+{
+    /// <summary>
+    /// Fetch the profile for a resource owner. Will compile a list of departments the person has responsibilities in and which is relevant.
+    /// </summary>
+    public class GetPersonRelevantOrgUnits : IRequest<QueryRangedList<QueryRelevantOrgUnit>>
+    {
+        public GetPersonRelevantOrgUnits(PersonId personId)
+        {
+            PersonId = personId;
+        }
+
+        /// <summary>
+        /// Mail or azure unique id
+        /// </summary>
+        public PersonId PersonId { get; }
+        public ODataQueryParams Query { get; set; } = new();
+
+        public GetPersonRelevantOrgUnits WithQuery(ODataQueryParams query)
+        {
+            Query = query;
+            return this;
+        }
+
+
+
+        public class Handler : IRequestHandler<GetPersonRelevantOrgUnits, QueryRangedList<QueryRelevantOrgUnit>>
+        {
+            private readonly IFusionProfileResolver profileResolver;
+            private readonly IFusionRolesClient rolesClient;
+            private readonly IMediator mediator;
+            private readonly IMemoryCache memoryCache;
+            private readonly HttpClient lineOrgClient;
+
+            public Handler(IFusionProfileResolver profileResolver, IFusionRolesClient rolesClient, IMediator mediator, IMemoryCache memoryCache, IHttpClientFactory httpClientFactory)
+            {
+                this.profileResolver = profileResolver;
+                this.rolesClient = rolesClient;
+                this.mediator = mediator;
+                this.memoryCache = memoryCache;
+                lineOrgClient = httpClientFactory.CreateClient(IntegrationConfig.HttpClients.ApplicationLineOrg());
+            }
+
+            public async Task<QueryRangedList<QueryRelevantOrgUnit>> Handle(GetPersonRelevantOrgUnits request, CancellationToken cancellationToken)
+            {
+                var orgUnits = await GetOrgUnitsAsync();
+
+                var relevantOrgUnits = await QueryAllRelevantOrgUnitsForUser(request, cancellationToken, orgUnits);
+                var filteredOrgUnits = ApplyOdataFilters(request.Query.Filter, relevantOrgUnits);
+
+                var totalCount = relevantOrgUnits.Count;
+
+                var skip = request.Query.Skip.GetValueOrDefault(0);
+                var take = request.Query.Top.GetValueOrDefault(totalCount);
+
+                var pagedQuery = QueryRangedList.FromItems(filteredOrgUnits, filteredOrgUnits.Count, skip);
+
+                return pagedQuery;
+            }
+
+            private static List<QueryRelevantOrgUnit> ApplyOdataFilters(ODataExpression filter, List<QueryRelevantOrgUnit> orgUnits)
```

Comment:
Tried to use the odata query expression builder for this? The sql is provided through Linq anyways.. 
There is a caveat tho, which is case insensitivity. This was done by always using `string.equals(string, string, IgnoreCase...)`. However at one point EF started throwing exceptions here, saying the cultureinfo.ordinalignorecase was not supported, as this was handled by the table collation. To solve this so it works with in memory collections and sql driver, there is a property one can use when mapping filter fields.. The first link shows a test where this has been done.



https://github.com/equinor/fusion-libraries/blob/2a2127950964b027baff348c6fc71a3be5b9f77a/src/Web/tests/Fusion.AspNetCore.Tests/OdataFilterTests.cs#L125

https://github.com/equinor/fusion-libraries/blob/2a2127950964b027baff348c6fc71a3be5b9f77a/src/Web/tests/Fusion.AspNetCore.Tests/OdataFilterTests.cs#L332